crypto/x509/v3_ncons.c

```diff
@@ -782,23 +783,40 @@ static int nc_email(ASN1_IA5STRING *eml, ASN1_IA5STRING *base)
 static int nc_uri(ASN1_IA5STRING *uri, ASN1_IA5STRING *base)
 {
     const char *baseptr = (char *)base->data;
-    const char *hostptr = (char *)uri->data;
-    const char *p = ia5memchr(uri, (char *)uri->data, ':');
+    const char *hostptr;
+    const char *p;
+    char *uri_copy;
+    char *scheme;
+    char *host;
     int hostlen;
 
-    /* Check for foo:// and skip past it */
-    if (p == NULL
-            || IA5_OFFSET_LEN(uri, p) < 3
-            || p[1] != '/'
-            || p[2] != '/')
+    if (!(uri_copy = OPENSSL_strndup((const char *) uri->data, uri->length)))
+        return X509_V_ERR_UNSPECIFIED;
+
+    if (!OSSL_parse_url(uri_copy, &scheme, NULL, &host, NULL, NULL, NULL, NULL, NULL)) {
+        OPENSSL_free(uri_copy);
         return X509_V_ERR_UNSUPPORTED_NAME_SYNTAX;
-    hostptr = p + 3;
+    }
 
-    /* Determine length of hostname part of URI */
+    /* Make sure the scheme and the host are there */
+    if (*scheme == '\0' || *host == '\0') {
+        OPENSSL_free(host);
+        OPENSSL_free(scheme);
+        OPENSSL_free(uri_copy);
+        return X509_V_ERR_UNSUPPORTED_NAME_SYNTAX;
+    }
 
-    /* Look for a port indicator as end of hostname first */
+    /* Determine the start of the hostname */
+    hostptr = strstr((const char *) uri->data, host);
```

Comment:
This part looks dubious to me. What is it supposed to do, any why is it needed?
Having parsed the URI, why are you looking at the raw URI again???

Reply:
Eh, I think I misunderstood your comment -- will rework the function to get rid of all IA5 function calls.

There seems to be an issue with CI: https://github.com/wireapp/openssl/actions/runs/12993398457/job/36235528380

It looks to me that the `no-http` config makes `OSSL_parse_url` unavailable (it lives under `crypto/http`), and I'm not sure how to address that.